python/packages/ai/teams/app.py

```diff
@@ -664,115 +664,90 @@ async def on_turn(self, context: TurnContext):
-
-            await state.load(context, self._options.storage)
-            state.temp.input = context.activity.text
-
-            # authentication
-            if (
-                self.options.auth
-                and self._auth
-                and (
-                    (
-                        (
-                            isinstance(self.options.auth.auto, bool)
-                            and self.options.auth.auto is True
-                        )
-                        or (
-                            callable(self.options.auth.auto)
-                            and self.options.auth.auto(context) is True
-                        )
-                    )
-                    or IN_SIGN_IN_KEY in state.user
-                )
-            ):
-                key: Optional[str] = (
-                    state.user[IN_SIGN_IN_KEY]
-                    if IN_SIGN_IN_KEY in state.user
-                    else self.options.auth.default
-                )
-
-                if key is not None:
-                    state.user[IN_SIGN_IN_KEY] = key
-                    res = await self._auth.sign_in(context, state, key=key)
-
-                    if res.status == "complete":
-                        del state.user[IN_SIGN_IN_KEY]
-
-                    if res.status == "pending":
-                        await state.save(context, self._options.storage)
-                        return
-
-                    if res.status == "error" and res.reason != "invalid-activity":
-                        del state.user[IN_SIGN_IN_KEY]
-                        raise ApplicationError(f"[{res.reason}] => {res.message}")
-
-            # run before turn middleware
-            for before_turn in self._before_turn:
-                is_ok = await before_turn(context, state)
-
-                if not is_ok:
-                    await state.save(context, self._options.storage)
-                    return
-
-            # download input files
-            if (
-                self._options.file_downloaders is not None
-                and len(self._options.file_downloaders) > 0
-            ):
-                input_files = state.temp.input_files if state.temp.input_files is not None else []
-                for file_downloader in self._options.file_downloaders:
-                    files = await file_downloader.download_files(context)
-                    input_files.append(files)
-                state.temp.input_files = input_files
-
-            # run activity handlers
-            is_ok, matches = await self._on_activity(context, state)
+            await self._start_typing(context)
+            await self._remove_mentions(context)
+            state = await self._initialize_state(context)
+            if not await self._run_before_turn_middleware(context, state):
+                return
 
+            await self._handle_file_downloads(context, state)
+            is_ok, matches = await self._run_activity_handlers(context, state)
```

Comment:
we can likely keep this section of running activity handlers the same - function extracted is a bit redundant